Rebuilt long beer example.

oldskeptic opened the pull request:
This refactors the long ACME Brewery according to all the changes in SOSA and follows the new file layout.

oldskeptic committed:
Rebuilt long beer example.

oldskeptic committed:
It's sosa:SamplingProcedure, not sosa:samplingProcedure dummy\!

dr-shorthair committed:
Merge branch 'gh-pages' into LongExampleAcme

dr-shorthair commented:
Preview: https://raw.githack.com/w3c/sdw-sosa-ssn/LongExampleAcme/ssn/index.html#beerTemp

dr-shorthair commented:
**C.1.1 Packaging and initial storage**

1.  Is `<acmePorterSixPack>` an individual or a class? 
In the first mention `<acmePorterSixPack> a beer:Porter, gs1:Product, schema:Product .` it is an individual, unless `beer:Porter`, `gs1:Product`, `schema:Product` are sub-classes (rather than _instances_) of `owl:Class`.  But a few lines lower down `<10/PO202402/21/0001/acmePorterSixPack> rdfs:subClassOf <acmePorterSixPack> .` it behaves like a class. 
2. It looks like you reused the classes from [8.12 Complex devices](https://w3c.github.io/sdw-sosa-ssn/ssn/#ModelComplexSystem) e.g. `sensor:IBS-TH2-Plus-H` - good! However, you have tied `<12345/HumiditySensor>` and `<12345/TemperatureSensor>` to `<12345/someTH2>` using `sosa:hosts` which makes latter a _Platform_. In [8.12 Complex devices](https://w3c.github.io/sdw-sosa-ssn/ssn/#ModelComplexSystem) it is a _System_ with sub-systems. 
3. Could `<00001/packingSixPack>` be modeled as a `sosa:Deployment`, either in addition to or instead of `prov:Activity`?

maximelefrancois86 reviewed (commented):
broken Fig 52 ?

oldskeptic committed:
Clean up warts according to @dr-shorthair's comments

oldskeptic committed:
Merge branch 'LongExampleAcme' of https://github.com/w3c/sdw-sosa-ssn into LongExampleAcme

oldskeptic commented:
> 1. Is `<acmePorterSixPack>` an individual or a class?
>    In the first mention `<acmePorterSixPack> a beer:Porter, gs1:Product, schema:Product .` it is an individual, unless `beer:Porter`, `gs1:Product`, `schema:Product` are sub-classes (rather than _instances_) of `owl:Class`.  But a few lines 
lower down `<10/PO202402/21/0001/acmePorterSixPack> rdfs:subClassOf <acmePorterSixPack> .` it behaves like a class.

Both. This is a untidy corner of both schema.org and gs1 product. In Page 115 of the [GS1 Digital Link Standard](https://www.gs1.org/docs/Digital-Link/GS1_Digital_link_Standard_i1.1.pdf), it must be a class because of the relationships between lots / serial numbers. It's an instance of [schema:IndividualProduct](https://schema.org/IndividualProduct) in schema.org land. The problem is also reviewed in gs1/WebVoc#38. I'm comfortable leaving it as both in this particular example, I added a [schema:IndividualProduct](https://schema.org/IndividualProduct) type to make the distinction clear.

Commercial implementation (me) run into this issue every day. Do you think that a short explanatory blurb would be beneficial in the long example?

> 2. It looks like you reused the classes from [8.12 Complex devices](https://w3c.github.io/sdw-sosa-ssn/ssn/#ModelComplexSystem) e.g. `sensor:IBS-TH2-Plus-H` - good! However, you have tied `<12345/HumiditySensor>` and `<12345/TemperatureSensor>` to `<12345/someTH2>` using `sosa:hosts` which makes latter a _Platform_. In [8.12 Complex devices](https://w3c.github.io/sdw-sosa-ssn/ssn/#ModelComplexSystem) it is a _System_ with sub-systems.

Fixed.

> 3. Could `<00001/packingSixPack>` be modeled as a `sosa:Deployment`, either in addition to or instead of `prov:Activity`?

A version of this is done in "C.1.4.1 Recording data and sensor tasking through sensor deployments".

oldskeptic commented:
> broken Fig 52 ?

It's rendering properly locally here. What breakage are you seeing?

dr-shorthair committed:
fix paths for code and image inclusions

dr-shorthair committed:
Merge branch 'gh-pages' into LongExampleAcme

dr-shorthair committed:
Merge branch 'gh-pages' into LongExampleAcme

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+    <div>
+     <p>The following is a detailed example where a IBS-TH2 sensor is used to monitor a crate (carton) of beer from the time of its
+     packaging at a brewery up until its display in the supermarket cooler where it is purchased.  Beer degrades both
+     at high and low temperatures.  This requires that it be kept cool during storage and shipping to preserve the
```
```suggestion
     at high and low temperatures.  This requires it to be kept cool during storage and shipping to preserve the
```

TallTed replied:
```suggestion
     packaging at a brewery up until its display in the supermarket cooler where it is purchased.  Beer degrades at
     both high and low temperatures.  This requires that its temperature be controlled during storage and shipping to preserve the
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     <p>The following is a detailed example where a IBS-TH2 sensor is used to monitor a crate (carton) of beer from the time of its
+     packaging at a brewery up until its display in the supermarket cooler where it is purchased.  Beer degrades both
+     at high and low temperatures.  This requires that it be kept cool during storage and shipping to preserve the
+     product. Humidity has no effect the beer owing to sealed containers through excess humidity will weaken cardboard and degrade
```
```suggestion
     product. Humidity has no effect on the beer, owing to sealed containers, though excess humidity will weaken cardboard and degrade
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     packaging at a brewery up until its display in the supermarket cooler where it is purchased.  Beer degrades both
+     at high and low temperatures.  This requires that it be kept cool during storage and shipping to preserve the
+     product. Humidity has no effect the beer owing to sealed containers through excess humidity will weaken cardboard and degrade
+     the adhesives potentially resulting in product breakage.  Brewers will on occasion insert a temperature logger into
```
```suggestion
     the adhesives, potentially resulting in product breakage.  Brewers will on occasion insert a temperature logger into
```

TallTed replied:
```suggestion
     adhesives, potentially resulting in product breakage.  Brewers will sometimes insert a temperature logger into
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     at high and low temperatures.  This requires that it be kept cool during storage and shipping to preserve the
+     product. Humidity has no effect the beer owing to sealed containers through excess humidity will weaken cardboard and degrade
+     the adhesives potentially resulting in product breakage.  Brewers will on occasion insert a temperature logger into
+     their shipping process as part of their quality control program and we present such a hypothetical situation here.<p>
```
```suggestion
     their shipping process as part of their quality control program. We present such a hypothetical situation here.<p>
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     the adhesives potentially resulting in product breakage.  Brewers will on occasion insert a temperature logger into
+     their shipping process as part of their quality control program and we present such a hypothetical situation here.<p>
+
+     <p>This example is of interest because it involves the repurposing of a commodity, consumer-grade <a href="#SOSAPlatform">sosa:Platform</a> to a
```
```suggestion
     <p>This example is of interest because it involves the repurposing of a commodity, a consumer-grade <a href="#SOSAPlatform">sosa:Platform</a> deployed to address a
```

TallTed replied:
```suggestion
     <p>This example is interesting because it repurposes a commodity, consumer-grade <a href="#SOSAPlatform"><code>sosa:Platform</code></a> to address a
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     their shipping process as part of their quality control program and we present such a hypothetical situation here.<p>
+
+     <p>This example is of interest because it involves the repurposing of a commodity, consumer-grade <a href="#SOSAPlatform">sosa:Platform</a> to a
+     domain-specific problem, the movement of a platform from environment to environment and the possible ambiguity of data-ownership in each of these environments. It also demonstrates the location 
```
```suggestion
     domain-specific problem, the movement of a platform from environment to environment, and the possible ambiguity of data-ownership in each of these environments. It also demonstrates the location 
```

TallTed replied:
```suggestion
     domain-specific problem: the movement of a platform from one environment to another, with possible data-ownership ambiguity in each environment. It also demonstrates the location 
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     the adhesives potentially resulting in product breakage.  Brewers will on occasion insert a temperature logger into
+     their shipping process as part of their quality control program and we present such a hypothetical situation here.<p>
+
+     <p>This example is of interest because it involves the repurposing of a commodity, consumer-grade <a href="#SOSAPlatform">sosa:Platform</a> to a
```
This is a rather long sentence. Consider using bullet points.

ldesousa commented on ssn/rdf/examples/Beer-Packaging-IBS-TH2.ttl:
```diff
@@ -0,0 +1,111 @@
+# identifier to highlight that this is a physical instance of the product.
+ 
+<10/PO202402/21/0001/acmePorterSixPack> a schema:IndividualProduct;
+    rdfs:subClassOf <acmePorterSixPack>;
```
```suggestion
    a <acmePorterSixPack>;
```

ldesousa commented on ssn/rdf/examples/Beer-Packaging-IBS-TH2.ttl:
```diff
@@ -0,0 +1,111 @@
+    sosa:observedProperty sosa-env:AmbientTemperature ;
+    sosa:hasUltimateFeatureOfInterest <10/PO202402/21/0001/acmePorterSixPackBeerSample> ; 
+    sosa:madeBySensor <12345/TemperatureSensor>  ;
+    sosa:hasFeatureOfInterest <10/PO202402/21/0001/acmePorterSixPackAirSample> ;
```
The individual `10/PO202402/21/0001/acmePorterSixPackAirSample` does not exist in the scope of this RDF document.

ldesousa commented on ssn/rdf/examples/Beer-Packaging-IBS-TH2.ttl:
```diff
@@ -0,0 +1,111 @@
+    sosa:observedProperty sosa-env:AmbientHumidity;
+    sosa:hasUltimateFeatureOfInterest <0001/ProductPackaging>;
+    sosa:madeBySensor <12345/HumiditySensor>  ;
+    sosa:hasFeatureOfInterest <10/PO202402/21/0001/acmePorterSixPackAirSample> ;
```
Same as above.

ldesousa commented on ssn/rdf/examples/Beer-Packaging-IBS-TH2.ttl:
```diff
@@ -0,0 +1,111 @@
+<1a/observation> a sosa:Observation ;
+    rel:own <acmeBreweryCo> ;
+    sosa:observedProperty sosa-env:AmbientTemperature ;
+    sosa:hasUltimateFeatureOfInterest <10/PO202402/21/0001/acmePorterSixPackBeerSample> ; 
```
```suggestion
    sosa:hasUltimateFeatureOfInterest <10/PO202402/21/0001/acmePorterSixPackBeer> ; 
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+   within a context <i>not necessarily intended by the original sensor design itself</i>.</p>
+   
+   <p>Physically, both &lt;12345/HumiditySensor&gt; and &lt;12345/TemperatureSensor&gt; are monitoring the ambient air within the
+   specific Porter carton which we define as <small>&lt;10/PO202402/21/0001/acmePorterSixPackAirSample&gt;</small> which resolves the issue of the
```
That individual is not declared in the RDF document above. Missing?

Also, the term "sample" use here is not the best, as in SOSA/SSN `Sample` is the outcome of a `Sampling`.

TallTed replied:
```suggestion
   specific Porter carton which we define as <small>&lt;10/PO202402/21/0001/acmePorterSixPackAirSample&gt;</small>, which resolves the issue of the
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+   then targets the beer temperature within that carton and the humidity the cardboard of the carton was subject to:</p>
+   <pre class="example turtle" data-include="./rdf/examples/Beer-FeatureOfInterest-IBS-TH2.ttl" data-include-format="text"></pre>
+
+   <p>The Sosa ontology takes a &quot;feature and property&quot; approach that is then referenced by observations. The
```
I understand the metaphor of the air sample to an extent, but really the beer sample. Isn't the air in the cartoon a proxy for the beer inside the bottles? Why is the "virtual sample" necessary?

And again, the air in the cartoon is not exactly a sample, according to SOSA semantics. Rather it is a proximate feature of interest.

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     </div> 
+      <p>After being
+     chilled in the brewery cooler, the beer carton is then loaded on one of the Acme delivery trucks for shipment to the retail
+     store.  The truck's cooling management system is recording the sensor readings for both the shipping company and the brewery.
```
```suggestion
     store.  The cooling management system of the truck is recording the sensor readings for both the shipping company and the brewery.
```

TallTed replied:
```suggestion
     store.  The cooling management system of the truck records the sensor readings for both the shipping company and the brewery.
```

ldesousa commented on ssn/rdf/examples/Beer-Shipping-IBS-TH2.ttl:
```diff
@@ -0,0 +1,72 @@
+#
+# Product is loaded on the truck
+#
+<67a/observation> a sosa:Observation, prov:Entity ;
```
There are multiple individuals used in this document that are never declared: 
- `10/PO202402/21/0001/BeerTemperature`
- `10/PO202402/21/0001/acmePorterSixPackBeerSample`
- `10/PO202402/21/0001/acmePorterSixPackAirSample`
- `12345/HumiditySensor`
- `12345/TemperatureSensor`

Was the idea here to provide fragments of a single RDF document? As it is that is not apparent.

ldesousa commented on ssn/rdf/examples/Beer-Shipping-IBS-TH2.ttl:
```diff
@@ -0,0 +1,72 @@
+    prov:startedAtTime "2024-02-22T04:15:05Z"^^xsd:dateTime ;
+    prov:endedAtTime "2024-02-22T05:55:38Z"^^xsd:dateTime ;
+    prov:wasStartedBy <driver> ;
+    prov:atLocation <gpsLocation> .
```
```suggestion
    prov:atLocation <123/gpsLocation> .
```

ldesousa commented on ssn/rdf/examples/Beer-Shipping-IBS-TH2.ttl:
```diff
@@ -0,0 +1,72 @@
+# This is the output from the truck's GPS receiver. We only get coordinates
+# from it.
+
+<gpsLocation> a prov:Location, geosparql:Geometry; 
```
```suggestion
<123/gpsLocation> a prov:Location, geosparql:Geometry; 
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+    location of the carton of beer does not effect the process because the platform monitors the inside of the carton itself. Modeling
+    its location and the process of (un)loading of the carton from the truck is done through other rdf nodes.</p> 
+    <p>The truck on-board monitoring system does report a GPS geometry as a Location. This is the location at which the data was
+    recorded from the platform. It is conceivable that the geometry node is shared with a vehicle tracking system rdf representation
```
```suggestion
    recorded from the platform. It is conceivable for the geometry node to be shared with a vehicle tracking system RDF representation
```

TallTed replied:
```suggestion
    recorded by the platform. It is conceivable for the geometry node to be shared with an RDF representation of the vehicle tracking system
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+    its location and the process of (un)loading of the carton from the truck is done through other rdf nodes.</p> 
+    <p>The truck on-board monitoring system does report a GPS geometry as a Location. This is the location at which the data was
+    recorded from the platform. It is conceivable that the geometry node is shared with a vehicle tracking system rdf representation
+    or the vehicle's delivery scheduling application but it is not mandated by Sosa.</p> 
```
```suggestion
    or the delivery scheduling application of the vehicle, but that is not mandated by SOSA.</p> 
```

TallTed replied:
```suggestion
    or the delivery scheduling application of the vehicle, but this is not mandated by SOSA.</p>
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+
+     <p class="note">Locations of Platform, Sensors, and measured samples are often conflated in
+     non-semantically enabled systems and the semantics often implicitly assumed by the application. The deep semantic modeling
+     within Sosa makes no such implicit assumptions and locations can be assigned to all elements independently.</p>
```
```suggestion
     within SOSA makes no such implicit assumptions and locations can be assigned to all elements independently.</p>
```

TallTed replied:
```suggestion
     within SOSA makes no such implicit assumptions, and locations can be assigned independently to each element.</p>
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      <p>When received at the retail store from the delivery company, the beer is displayed for sale in retail coolers which log
+      the sensors, the data being held by the retail store. We notice again the semantics of location and containment; the sensor is
+      contained within the carton which is contained within the cooler which is itself located within the store. However, none of this
+      semantic is recorded. The only location activity is 
```
```suggestion
      semantic is recorded. The only relevant location is 
```

TallTed replied:
```suggestion
      semantics are recorded. The only relevant location is 
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      the sensors, the data being held by the retail store. We notice again the semantics of location and containment; the sensor is
+      contained within the carton which is contained within the cooler which is itself located within the store. However, none of this
+      semantic is recorded. The only location activity is 
+      the activity is taking place within the supermarket cooler; not the supermarket itself. </p>
```
```suggestion
      the activity taking place within the supermarket cooler; not the supermarket itself. </p>
```

TallTed replied:
```suggestion
     the supermarket cooler; not the supermarket itself.</p>
```

ldesousa commented on ssn/rdf/examples/Beer-Supermarket-IBS-TH2.ttl:
```diff
@@ -0,0 +1,58 @@
+# changed and observation data is now only being recorded / owned by supermarket.
+#
+    
+<98a/observation> rdf:type sosa:Observation,prov:Entity ;
```
Again here the issue with undeclared individuals.

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      
+      <h5>Recording data and sensor tasking through sensor deployments</h5>
+      
+      <p>Sosa provides a notion of a <a href="#SOSADeployment">Deployment</a> class. This permits the tasking
```
```suggestion
      <p>SOSA provides the notion of a <a href="#SOSADeployment">Deployment</a> class. This permits the tasking
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      
+      <p>Sosa provides a notion of a <a href="#SOSADeployment">Deployment</a> class. This permits the tasking
+      or installation of sensors to different environments, purposes, locations or
+      leases. Intended to link Platforms to System this allows the application of different semantics to different situations,
```
```suggestion
      leases. Intended to link Platforms to System, it allows the application of different semantics to different situations,
```

TallTed replied:
```suggestion
      leases. Intended to link Platforms to Systems, it allows the application of different semantics to different situations,
```

ldesousa commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      <p>Sosa provides a notion of a <a href="#SOSADeployment">Deployment</a> class. This permits the tasking
+      or installation of sensors to different environments, purposes, locations or
+      leases. Intended to link Platforms to System this allows the application of different semantics to different situations,
+      for example by recording different configuration parameters, thought deployments may be concurrent. Reusing the example of [[[#beerTemp]]], the sensor can be seen to be concurrently deployed within the
```
```suggestion
      for example by recording different configuration parameters, though deployments may be concurrent. Reusing the example of [[[#beerTemp]]], the sensor can be seen to be concurrently deployed within the
```

TallTed replied:
```suggestion
      by recording different configuration parameters, for example, though deployments may be concurrent. Reusing the example of [[[#beerTemp]]], the sensor can be seen to be concurrently deployed within one
```

ldesousa commented on ssn/rdf/examples/Beer-PlatformDeployment-IBS-TH2.ttl:
```diff
@@ -0,0 +1,67 @@
+    prov:wasAssociatedWith <alice> ;
+    sosa:deployedOnPlatform <10/PO202402/21/0001/acmePorterSixPack> ;
+    sosa:deployedSystem <12345/someTH2> ;
+    sosa:forProperty sosa-env:AmbientHumidity, <10/PO202402/21/0001/BeerTemperature> ;
```
This predicated is not meant to be used with `sosa:Deployment`. The domain is declared as `sosa:Procedure` or `sosa:System`. While domain and ranges are not forcefull (`domainIncludes`) it would be better if the examples stick to the spec.

ldesousa commented on ssn/rdf/examples/Beer-SampleCollections-IBS-TH2.ttl:
```diff
@@ -0,0 +1,36 @@
+
+<10/PO202402/21/0001/acmePorterSixPackLabSample> a sosa:Sample;
+    rdfs:label "A sample of beer within the six pack"@en ;
+    sosa:isResultOf <ACMEqualityAssuranceProgram>;
```
`sosa:isResultOf` is meant to be used with `sosa:Execution` as range. Therefore `ACMEqualityAssuranceProgram` *should* be a `sosa:Sampling`. 

I think it is best if the examples keep to the spirit of the spec. While this creative use of predicates is not illegal it adds unnecessary entropy in my view.

oldskeptic replied:
OK.

ldesousa reviewed (dismissed):
Besides the individual comments there are two broad aspects to consider:
1. Use of predicates: better to stick to the intended use.
2. TTL documents: each should be a contained asset, including all individual declarations necessary. In alternative encapsulate recurrent individuals (e.g. sensors) in their own document, to be then "imported" with `@prefix`.

dr-shorthair committed:
Merge branch 'gh-pages' into LongExampleAcme

oldskeptic commented:
To be moved to a markdown example.

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     domain-specific problem, the movement of a platform from environment to environment and the possible ambiguity of data-ownership in each of these environments. It also demonstrates the location 
+     ambiguity that can sometimes be present in modeling sensor readings.</p>
+
+     <p class="note">The use of long identifiers is made for readability purposes and is non-normative.</p>
```
```suggestion
     <p class="note">The use of long identifiers is for readability purposes and is non-normative.</p>
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+
+     <p>This example is of interest because it involves the repurposing of a commodity, consumer-grade <a href="#SOSAPlatform">sosa:Platform</a> to a
+     domain-specific problem, the movement of a platform from environment to environment and the possible ambiguity of data-ownership in each of these environments. It also demonstrates the location 
+     ambiguity that can sometimes be present in modeling sensor readings.</p>
```
```suggestion
     ambiguity that can sometimes be present when modeling sensor readings.</p>
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     
+     <p>Alice works for Acme Brewery Co., she procures an InkBird IBS-TH2 Platform (Serial Number: 12345) and places it inside a beer carton with six
+     bottles (a &quot;six-pack&quot;) of Acme Brewery's famous Porter beers with a product lot code of PO202402. She seals the packaging and places it inside of Acme
+     Brewery's cooler while awaiting shipment.</p>
```
```suggestion
     <p>Alice works for Acme Brewery Co. She procures an InkBird IBS-TH2 Platform (Serial Number: 12345) and places it inside a beer carton with six
     bottles (a &quot;six-pack&quot;) of Acme Brewery's famous Porter beers with a product lot code of PO202402. She seals the package and places it inside of Acme
     Brewery's cooler to await shipment.</p>
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     <p>We model the packaging activity and the placement of the Platform into the
+     crate before it is sealed. The platform is thus monitoring the ambient air temperature and the relative humidity from within
+     the beer crate every few minutes. In the following snippet we instantiate both the IBS-TH2 Platform, the carton and the
+     activity that represents the packaging of the beer and the creation of the beer carton as a product:<p>
```
```suggestion
     <p>We model the packaging activity and the placement of the Platform into the
     crate before it is sealed. The platform monitors the ambient air temperature and the relative humidity within
     the beer crate every few minutes. In the following snippet, we instantiate the IBS-TH2 Platform, the carton, and the
     activity that represents the packaging of the beer and the creation of the beer carton as a product:<p>
```

oldskeptic commented:
...it was the night before Christmas and all through the repo, nothing was stirring, not even an issue...

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+   temperature within each beer vessel. Modeling the full thermodynamics activity relating one measurement to the other is beyond the
+   scope of this document but could be implemented as a second order &quot;virtual&quot; <a href="#SOSASensor">sosa:Sensor</a> or a
+   sophisticated <a href="#SOSAProcedure">sosa:Procedure</a>. Thus the use of the <a href="#SOSAhasUltimateFeatureOfInterest">sosa:hasUltimateFeatureOfInterest</a> property allows for flexibility in the interpretation of the measurement
+   within a context <i>not necessarily intended by the original sensor design itself</i>.</p>
```
```suggestion
   <p>In the example above, concurrent use of the <a href="#SOSAhasFeatureOfInterest"><code>sosa:hasFeatureOfInterest</code></a> and <a
   href="#SOSAhasUltimateFeatureOfInterest"><code>sosa:hasUltimateFeatureOfInterest</code></a> properties is made to account for the repurposing of a generic
   sensor. The actual measurement performed by the platform is the air temperature within the carton, as a proxy for the beer
   temperature within each beer vessel. Modeling the full thermodynamic activity relating one measurement to the other is beyond the
   scope of this document, but could be implemented as a second order &quot;virtual&quot; <a href="#SOSASensor"><code>sosa:Sensor</code></a> or a
   sophisticated <a href="#SOSAProcedure"><code>sosa:Procedure</code></a>. Thus the use of the <a href="#SOSAhasUltimateFeatureOfInterest"><code>sosa:hasUltimateFeatureOfInterest</code></a> property allows for flexibility in interpretation of the measurement
   within a context <i>not necessarily intended by the original sensor design</i>.</p>
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+   specific Porter carton which we define as <small>&lt;10/PO202402/21/0001/acmePorterSixPackAirSample&gt;</small> which resolves the issue of the
+   location of the measurement when the carton itself is in motion. The <a href="#SOSAhasUltimateFeatureOfInterest">sosa:hasUltimateFeatureOfInterest</a> property
+   then targets the beer temperature within that carton and the humidity the cardboard of the carton was subject to:</p>
+   <pre class="example turtle" data-include="./rdf/examples/Beer-FeatureOfInterest-IBS-TH2.ttl" data-include-format="text"></pre>
```
```suggestion
   location of the measurement when the carton itself is in motion. The <a href="#SOSAhasUltimateFeatureOfInterest"><code>sosa:hasUltimateFeatureOfInterest</code></a> property
   targets the beer temperature within that carton and the humidity the cardboard of the carton was subjected to:</p>
   <pre class="example turtle" data-include="./rdf/examples/Beer-FeatureOfInterest-IBS-TH2.ttl" data-include-format="text"></pre>
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+   individual containers ( <small>&lt;10/PO202402/21/0001/acmePorterSixPackBeerSample&gt; </small>) within the carton and the relative humidity that
+   the packaging ( <small>&lt;0001/ProductPackaging&gt; </small>) is being exposed to. The beer &quot;sample&quot; in this case is virtual; the node
+   serves only to semantically link the observation to the beer liquid itself without any physical sample being taken. A more
+   detailed example with physical beer samples is listed in [[[#beerCollections]]].</p>     
```
```suggestion
   <p>The SOSA ontology takes a &quot;feature and property&quot; approach that is referenced by observations. The
   feature of interest &mdash; in this case, the ambient air within the carton
   (<small>&lt;10/PO202402/21/0001/acmePorterSixPackAirSample&gt;</small>) &mdash; is
   the feature that the Platform is concerned with. However, it is really a proxy for both the temperature of the beer within the
   individual containers (<small>&lt;10/PO202402/21/0001/acmePorterSixPackBeerSample&gt;</small>)
   within the carton and the relative humidity to which
   the packaging (<small>&lt;0001/ProductPackaging&gt;</small>) is being exposed. The beer &quot;sample&quot; in this case is virtual; the node
   serves only to semantically link the observation to the beer liquid itself without any physical sample being taken. A more
   detailed example with physical beer samples is found in [[[#beerCollections]]].</p>     
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     <div>
+      <figure id="InkBird_IBS_TH2-shipping" style="float: right; max-width: 30%; height: auto;vertical-align: top;margin: -10% 15px 0px 15px;">
+      <img src="./images/InkBird_IBS_TH2-shipping.png"> 
+      <figcaption>The beer carton is being moved from the brewery to the retail location.</figcaption> 
```
```suggestion
      <figcaption>The beer carton is moved from the brewery to the retail location.</figcaption> 
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      </figure> 
+     </div> 
+      <p>After being
+     chilled in the brewery cooler, the beer carton is then loaded on one of the Acme delivery trucks for shipment to the retail
```
```suggestion
      <p>After being
     chilled in the brewery cooler, the beer carton is loaded onto one of the Acme delivery trucks for shipment to the retail
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+     position information represents the location of the shipping truck itself without making reference to the beer carton or the
+     platform generating the readings. This is important in that the underlying properties and features referenced by the sensors
+     of the platform have not changed, even
+     through the beer carton and the platform have obviously changed physical location.</p>
```
```suggestion
     An on-board GPS unit annotates the activity of recording observations as the truck moves from location to location. This
     position information represents the location of the shipping truck itself without reference to the beer carton or the
     platform generating the readings. This is important, in that the underlying properties and features referenced by the sensors
     of the platform have not changed, even
     through the beer carton and the platform have obviously changed physical location.</p>
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+    <p>The sensors of the platform are reporting values for the same properties of the same features of interest; the physical
+    location of the carton of beer does not effect the process because the platform monitors the inside of the carton itself. Modeling
+    its location and the process of (un)loading of the carton from the truck is done through other rdf nodes.</p> 
+    <p>The truck on-board monitoring system does report a GPS geometry as a Location. This is the location at which the data was
```
```suggestion
    <p>The sensors of the platform report values for the same properties of the same features of interest. The physical
    location of the carton of beer does not affect the process because the platform monitors the inside of the carton itself; modeling
    its location and the process of (un)loading of the carton from the truck is done through other RDF nodes.</p> 
    <p>The on-board monitoring system of the truck does report a GPS geometry as a Location. This is the location at which the data was
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+    or the vehicle's delivery scheduling application but it is not mandated by Sosa.</p> 
+
+     <p class="note">Locations of Platform, Sensors, and measured samples are often conflated in
+     non-semantically enabled systems and the semantics often implicitly assumed by the application. The deep semantic modeling
```
```suggestion
     <p class="note">Locations of Platforms, Sensors, and measured samples are frequently conflated in
     non-semantically enabled systems and in the semantics often implicitly assumed by the application. The deep semantic modeling
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      </figure> 
+      <p>When received at the retail store from the delivery company, the beer is displayed for sale in retail coolers which log
+      the sensors, the data being held by the retail store. We notice again the semantics of location and containment; the sensor is
+      contained within the carton which is contained within the cooler which is itself located within the store. However, none of this
```
```suggestion
      <p>When received from the delivery company by the retail store, the beer is displayed for sale in retail coolers which log
      the sensors to the data being held by the retail store. We notice again the semantics of location and containment; the sensor is
      contained within the carton, which is contained within the cooler, which is itself located within the store. However, none of these
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      <h4>Notes on deployment and observation aggregation</h4>
+      
+      <p>In previous examples, <a href="https://www.w3.org/TR/prov-o/Activity">prov:Activity</a> was used as a means of aggregating
+      results as an archiving reception. Other mechanisms exists to define the purpose of the System and aggregate collections of
```
```suggestion
      results in an archiving reception. Other mechanisms exists to define the purpose of the System and to aggregate collections of
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      <h5>Recording data and sensor tasking through sensor deployments</h5>
+      
+      <p>Sosa provides a notion of a <a href="#SOSADeployment">Deployment</a> class. This permits the tasking
+      or installation of sensors to different environments, purposes, locations or
```
```suggestion
      or installation of sensors to different environments, purposes, locations, or
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      or installation of sensors to different environments, purposes, locations or
+      leases. Intended to link Platforms to System this allows the application of different semantics to different situations,
+      for example by recording different configuration parameters, thought deployments may be concurrent. Reusing the example of [[[#beerTemp]]], the sensor can be seen to be concurrently deployed within the
+      beer carton and multiple coolers / storage sites:</p>
```
```suggestion
      beer carton and multiple coolers or storage sites:</p>
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      <p>Sosa provides a lightweight capability to aggregate members for convenience. In previous examples observations where
+      clustered according to Deployment or recording Activity. In some circumstances it is useful to aggregate observations
+      according to an ad-hoc classification, here following the location storage of the carton while dispensing with geospatial
+      data:</p>
```
```suggestion
      <p>SOSA provides a lightweight capability to aggregate members for convenience. In previous examples, observations where
      clustered according to Deployment or recording Activity. In some circumstances, it is useful to aggregate observations
      according to an ad-hoc classification, here following the storage location of the carton while dispensing with geospatial
      data:</p>
```

TallTed commented on ssn/chapters/CompleteExamples.html:
```diff
@@ -1,9 +1,158 @@
+      <h5 id="beerSamples">Recording data using sample collection</h5> 
+    <p>A similar class <a href="#SOSASampleCollection">sosa:SampleCollection</a> allows for the aggregation of <a
+    href="#SOSASample">sosa:Sample</a>s, for example as part of a collection of samples taken as part of a
+    quality assurance program. In the case of Acme Brewery, a collection of beer samples could be represented in this manner:</p>       
```
```suggestion
    <p>A similar class &mdash; <a href="#SOSASampleCollection"><code>sosa:SampleCollection</code></a> &mdash; allows for the aggregation of <a
    href="#SOSASample"><code>sosa:Samples</code></a>, for example as part of a collection of samples taken as part of a
    quality assurance program. In the case of Acme Brewery, a collection of beer samples could be represented in this manner:</p>       
```

TallTed commented on ssn/rdf/examples/Beer-Full-IBS-TH2.ttl:
```diff
@@ -0,0 +1,230 @@
+    ex:deviceAddress "12:34:56:12:34:56" .
+
+<00001/packingSixPack> a prov:Activity, sosa:Deployment ;
+    rdfs:comment "When Alice packaged Porter bottles into the box, she added a InkBird logger to check that the beer wasn't getting too warm in transit and storage." ;    
```
```suggestion
    rdfs:comment "When Alice packaged Porter bottles into the box, she added an InkBird logger to check that the beer wasn't getting too warm in transit or storage." ;    
```

TallTed commented on ssn/rdf/examples/Beer-Packaging-IBS-TH2.ttl:
```diff
@@ -0,0 +1,111 @@
+    ex:deviceAddress "12:34:56:12:34:56" .
+
+<00001/packingSixPack> a prov:Activity ;
+    rdfs:comment "When Alice packaged Porter bottles into the box, she added a InkBird logger to check that the beer wasn't getting too warm in transit and storage." ;    
```
```suggestion
    rdfs:comment "When Alice packaged Porter bottles into the box, she added an InkBird logger to check that the beer wasn't getting too warm in transit and storage." ;    
```

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

oldskeptic committed:
Update ssn/rdf/examples/Beer-Packaging-IBS-TH2.ttl

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/rdf/examples/Beer-Full-IBS-TH2.ttl

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Update ssn/chapters/CompleteExamples.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

oldskeptic committed:
Partial move to markdown

oldskeptic committed:
Testing some markdown.

oldskeptic committed:
Fix main read-me file

dr-shorthair committed:
Fix link to examples folder in GitHub

dr-shorthair committed:
Link syntax

dr-shorthair committed:
Fixed image display syntax

dr-shorthair committed:
Use Markdown syntax for links - establishing the pattern

oldskeptic committed:
Partial fixes

dr-shorthair commented:
Unfortunately, GitHub markdown does not support `include` so you either have to copy the content over, else just provide a link.

oldskeptic commented:
> Unfortunately, GitHub markdown does not support `include` so you either have to copy the content over, else just provide a link.

Yes. Tinkering with a markdown generator. Will throw in the towel if results aren't good EOD.

oldskeptic committed:
Testing inline code

oldskeptic committed:
Move ACME example to sub-directory.

oldskeptic committed:
Added table of contents

oldskeptic committed:
Why do you hate me markdown.

oldskeptic committed:
Resolves discussion_r2610150362

oldskeptic commented:
> Besides the individual comments there are two broad aspects to consider:
> 
> 1. Use of predicates: better to stick to the intended use.
> 2. TTL documents: each should be a contained asset, including all individual declarations necessary. In alternative encapsulate recurrent individuals (e.g. sensors) in their own document, to be then "imported" with `@prefix`.

Fixed 1 and going to  push back on 2 since the length of the examples are getting too long.

oldskeptic committed:
Still in markdown hell.

dr-shorthair commented:
View the Long Beer Example in Markdown here: https://github.com/w3c/sdw-sosa-ssn/blob/LongExampleAcme/ssn/rdf/examples/ACME-Beer-Example.md

oldskeptic committed:
Partial work.

dr-shorthair committed:
Merge branch 'gh-pages' into LongExampleAcme

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-FeatureOfInterest-IBS-TH2.ttl:
```diff
@@ -0,0 +1,30 @@
+
+# A (virtual) sample of the beer from the six pack, implicitly assumes that the
+# sample is representative of the contents of all bottles. 
+<10/PO202402/21/0001/acmePorterSixPackBeerSample> a  sosa:FeatureOfInterest, sosa:Sample, beer:Porter ;
```
Whereas this might have been fine with SOSA v1.0, my understanding is that v2.0 tears appart both concepts. See the notes in https://w3c.github.io/sdw-sosa-ssn/ssn/#SOSASample.

dr-shorthair replied:
This is explained in https://w3c.github.io/sdw-sosa-ssn/ssn/#ModelDomainTypes 
<10/PO202402/21/0001/acmePorterSixPackBeerSample> is created as a `Sample` of 'all bottles'. 
Then when it is the subject of observation it becomes a `FeatureOfInterest`. 
The type evolves as it moves through the process.

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-FeatureOfInterest-IBS-TH2.ttl:
```diff
@@ -0,0 +1,30 @@
+# generic definition for the temperature of the sample.
+<10/PO202402/21/0001/BeerTemperature> a sosa:Property ;
+    qudt:hasUnit unit:DEG_C ;
+    sosa:isPropertyOf <10/PO202402/21/0001/acmePorterSixPackBeerSample> ;
```
This axiom leads to consider `acmePorterSixPackBeerSample` as a FoI, not as a Sample.

dr-shorthair replied:
That's OK. 
It sets the scene for an `Observation` with <10/PO202402/21/0001/BeerTemperature> as the `observedProperty` and <10/PO202402/21/0001/acmePorterSixPackBeerSample> as the `hasFeatureOfInterest`

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-Full-IBS-TH2.ttl:
```diff
@@ -0,0 +1,230 @@
+# identifier to highlight that the carton is a physical instance.
+ 
+<10/PO202402/21/0001/acmePorterSixPack> a schema:IndividualProduct;
+    rdfs:subClassOf <acmePorterSixPack>;
```
An individual that is simultâneously a class. Whereas this may be valid RDF, is this the kind of example we wish someone learning SOSA to deal with?

dr-shorthair replied:
I think the point here is that this is a reality that we have to deal with in the context of `schema:` and `GS1:` etc. 
Maybe add a #comment inline acknowledging this?

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-Full-IBS-TH2.ttl:
```diff
@@ -0,0 +1,230 @@
+
+# A (virtual) sample of the beer from the six pack, implicitly assumes that the
+# sample is representative from all bottles. 
+<10/PO202402/21/0001/acmePorterSixPackBeerSample> a  sosa:FeatureOfInterest, sosa:Sample, beer:Porter ;
```
This individual is defined in a previous document. On purpose?

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-Full-IBS-TH2.ttl:
```diff
@@ -0,0 +1,230 @@
+<12345/someTH2> a sensor:IBS-TH2-Plus, schema:IndividualProduct ;
+    rel:own <acmeBreweryCo> ; # Sensor may be returned.
+    rdfs:label "InkBird Sensor that Alice bought to track beer storage."@en ;
+    sosa:hasSubSystem <12345/HumiditySensor>, <12345/TemperatureSensor> ;
```
One can infer from this axiom that `someTH2` is a `sosa:System`, but why not explicitly declaring it as such?

dr-shorthair replied:
In fact it is a `sosa:Sensor` (subclass of `System`). 
It should be pointed out that `sensor:IBS-TH2-Plus` is defined in https://w3c.github.io/sdw-sosa-ssn/ssn/#ModelComplexSystem

ldesousa replied:
I had similar comments in the first review. This one more implicit aspect that seems unnecessary. A modular approach can be made explicit with a prefix pointing to the document where these individuals are declared.

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-Full-IBS-TH2.ttl:
```diff
@@ -0,0 +1,230 @@
+
+# These definitions look redundant; but they represent the specific, physical instantiation of the sensors.
+
+<12345/HumiditySensor> a sensor:IBS-TH2-Plus-H ;
```
One can infer these sensors to be also instances of `sosa:Sensor`, but I would recommend explicitly declaring them as such.

dr-shorthair replied:
`sensor:IBS-TH2-Plus-H` is defined in https://w3c.github.io/sdw-sosa-ssn/ssn/#ModelComplexSystem

ldesousa replied:
Same as above, I suggest creating an explicit reference to other documents with a `prefix` statement.

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-ObservationCollections-IBS-TH2.ttl:
```diff
@@ -0,0 +1,33 @@
+# Represent all data as Observation collections.
+
+<acmePorterSixPackObservations> a sosa:ObservationCollection;
+    sosa:hasMember <observation1a>, <observation1b>;
```
Suggest creating a list here.

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-Packaging-IBS-TH2.ttl:
```diff
@@ -0,0 +1,111 @@
+# identifier to highlight that this is a physical instance of the product.
+ 
+<10/PO202402/21/0001/acmePorterSixPack> a schema:IndividualProduct;
+    rdfs:subClassOf <acmePorterSixPack>;
```
Another issue with this formulation: `acmePorterSixPack` is never actually instanciated, it is always used as an individual. Then why add this complexity?

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-Packaging-IBS-TH2.ttl:
```diff
@@ -0,0 +1,111 @@
+
+# Alice packages the Beer.
+
+<12345/someTH2> a sensor:IBS-TH2-Plus, schema:IndividualProduct ;
```
Suggest declaring explicitly as `sosa:Sensor`. Same for the other devices declared below.

dr-shorthair replied:
` sensor:IBS-TH2-Plus` is defined in https://w3c.github.io/sdw-sosa-ssn/ssn/#ModelComplexSystem

ldesousa replied:
See below...

dr-shorthair commented:
@oldskeptic to tighten up the section at the top of the md document to more strongly draw attention to punning, merging OWL with schema.org and GS1 concerns. 
Work with @rob-metalinkage

ldesousa commented on ssn/rdf/examples/ACME-Beer/Beer-SampleCollections-IBS-TH2.ttl:
```diff
@@ -0,0 +1,36 @@
+    sosa:isResultOf <ACMEqualityAssuranceProgram>;
+    sosa:isResultOfUsedProcedure <beerSamplingProcedure> ;
+    ex:inside <0005/sampleFlask>;
+    sosa:isSampleOf <10/PO202402/21/0001/acmePorterSixPack> .
```
I might have already left this comment in a previous review. Looks like subject of this triple is defined in a different document. I suggest adding it to prefixes. Same for similar instances in the example.

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+standardized offerings need to be communicated in a way that is machine
+readable and other vocabularies are available to do so.
+
+From a procurement and product recall approach, vocabulatories are
```
```suggestion
From a procurement and product recall approach, vocabularies
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+differentiate between the product class and the specific instance of the
+product being consumed.  The ability to align Sosa and product vocabularies
+are beneficial in that it permits a unified description of the sensor.  This
+include modeling the sensor throughout its lifetime, announcing commercial
```
```suggestion
includes modeling the sensor throughout its lifetime, announcing commercial
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+availability, procurement, deployment, operation, decomissioning and
+disposal.
+
+To this end this example models IBS TH2 sensors using schema.org Product and
```
```suggestion
To this end this example models IBS TH2 sensors using [Schema.org](https://schema.org/) Product and
```

TallTed replied:
Any reason not to link directly?
```suggestion
To this end this example models IBS TH2 sensors using [Schema.org Product](https://schema.org/Product) and
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+Optimization (SOA) and consumer supply chain product identification using
+barcodes.
+
+Different design decisions also mean that design desision such as [punning](https://www.w3.org/2007/OWL/wiki/Punning)
```
```suggestion
Different design decisions also mean that design decisions such as [punning](https://www.w3.org/2007/OWL/wiki/Punning)
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+temperature and the relative humidity within the beer crate every few
+minutes.  In the following snippet, we instantiate the IBS-TH2 Platform, the
+carton, and the activity that represents the packaging of the beer and the
+creation of the beer carton as a product:
```
Anything missing after the colon?

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+In the example above, concurrent use of the 
+[sosa:hasFeatureOfInterest](https://www.w3.org/TR/vocab-ssn/#SOSAhasFeatureOfInterest) 
+and [sosa:hasUltimateFeatureOfInterest](https://www.w3.org/TR/vocab-ssn/#SOSAhasUltimateFeatureOfInterest) 
+properties is made to account for the repurposing of a generic sensor.  The
```
```suggestion
object properties is made to account for the repurposing of a generic sensor.  The
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+other is beyond the scope of this document, but could be implemented as a
+second order "virtual" [sosa:Sensor](https://www.w3.org/TR/vocab-ssn/#SOSASensor) or a sophisticated [sosa:Procedure](https://www.w3.org/TR/vocab-ssn/#SOSAProcedure).  Thus the use of the [sosa:hasUltimateFeatureOfInterest](https://www.w3.org/TR/vocab-ssn/#SOSAhasUltimateFeatureOfInterest) property allows for flexibility in interpretation of the measurement within a context *not necessarily inten ...
+   
+Physically, both <12345/HumiditySensor> and <12345/TemperatureSensor> are monitoring the ambient air within the specific Porter carton which we define as <10/PO202402/21/0001/acmePorterSixPackAirSample> which resolves the issue of the location of the measurement when the carton itself is in motion. The <a href="#SOSAhasUltimateFeatureOfInterest">`sosa:hasUltimateFeatureOfInterest`</a> property ta ...
```
```suggestion
Physically, both <12345/HumiditySensor> and <12345/TemperatureSensor> are monitoring the ambient air within the specific Porter carton which we define as <10/PO202402/21/0001/acmePorterSixPackAirSample> which resolves the issue of the location of the measurement when the carton itself is in motion. The <a href="#SOSAhasUltimateFeatureOfInterest">`sosa:hasUltimateFeatureOfInterest`</a> object property targets the beer temperature within that carton and the humidity the cardboard of the carton was subjected to:
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+proxy for both the temperature of the beer within the individual containers
+(<10/PO202402/21/0001/acmePorterSixPackBeerSample>) within the carton and
+the relative humidity to which the packaging (<0001/ProductPackaging>) is
+being exposed.  The beer "sample" in this case is virtual; the node serves
```
In my view this contradicts the definition of `sosa:Sample`.

dr-shorthair replied:
I see no problem with notional samples as part of the workflow.

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+
+After being chilled in the brewery cooler, the beer carton is then loaded on
+one of the Acme delivery trucks for shipment to the retail store.  The
+truck's cooling management system is recording the sensor readings for both
```
```suggestion
cooling management system of the truck is recording the sensor readings for both
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+truck itself without reference to the beer carton or the platform generating
+the readings.  This is important, in that the underlying properties and
+features referenced by the sensors of the platform have not changed, even
+through the beer carton and the platform have obviously changed physical
```
```suggestion
though the beer carton and the platform have obviously changed physical
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+the same features of interest; the physical location of the carton of beer
+does not effect the process because the platform monitors the inside of the
+carton itself.  Modeling its location and the process of (un)loading of the
+carton from the truck is done through other rdf nodes.  The truck on-board
```
```suggestion
carton from the truck is done through other RDF nodes.  The truck on-board
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+carton from the truck is done through other rdf nodes.  The truck on-board
+monitoring system does report a GPS geometry as a Location.  This is the
+location at which the data was recorded from the platform.  It is
+conceivable that the geometry node is shared with a vehicle tracking system
```
```suggestion
conceivable that the geometry node is shared with a vehicle tracking system.
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+monitoring system does report a GPS geometry as a Location.  This is the
+location at which the data was recorded from the platform.  It is
+conceivable that the geometry node is shared with a vehicle tracking system
+rdf representation or the vehicle's delivery scheduling application but it
```
```suggestion
RDF representation or the vehicle's delivery scheduling application but it
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+
+<img src="../../images/InkBird_IBS_TH2-receiving.png" width="25%" alt="The beer carton is stored in the display cooler for retail sale.">
+
+When received from the delivery company by the retail store, the beer is displayed for sale in retail coolers which log the sensors to the data being held by the retail store. We notice again the semantics of location and containment; the sensor is contained within the carton, which is contained within the cooler, which is itself located within the store. However, none of these semantic is record ...
```
```suggestion
When received from the delivery company by the retail store, the beer is displayed for sale in retail coolers which log the sensors to the data being held by the retail store. We notice again the semantics of location and containment; the sensor is contained within the carton, which is contained within the cooler, which is itself located within the store. However, none of these semantics are recorded. The only location activity is taking place within the supermarket cooler; not the supermarket itself. 
```

ldesousa commented on ssn/rdf/examples/ACME-Beer-Example.md:
```diff
@@ -0,0 +1,617 @@
+
+## Notes on deployment and observation aggregation
+ 
+In previous examples, <a href="https://www.w3.org/TR/prov-o/Activity">prov:Activity</a> was used as a means of aggregating results in an archiving reception. Other mechanisms exists to define the purpose of the System and to aggregate collections of results within specific contexts. 
```
```suggestion
In previous examples, <a href="https://www.w3.org/TR/prov-o/Activity">prov:Activity</a> was used as a means of aggregating results in an archiving reception. Other mechanisms exist to define the purpose of the System and to aggregate collections of results within specific contexts. 
```

ldesousa commented on ssn/rdf/examples/readme.md:
```diff
@@ -1,3 +1,55 @@
-# SOSA examples
+# Extended SOSA examples
+
+This folder contains long running examples previously contained within the RDF files in this directory correspond to the numbered examples in the [Semantic Sensor Network Ontology](https://www.w3.org/TR/vocab-ssn/) recommendation. 
```
```suggestion
This folder contains long running examples previously contained within the RDF files in this directory. They correspond to the numbered examples in the [Semantic Sensor Network Ontology](https://www.w3.org/TR/vocab-ssn/) recommendation. 
```

ldesousa commented on ssn/rdf/examples/readme.md:
```diff
@@ -1,3 +1,55 @@
+## Tracking Beer Temperature with IBS-TH2
+
+This example is documented in more detail in [ACME-Beer-Example.md](./ACME-Beer-Example.md) 
+An RDF file containing a [example ttl graph file](Beer-Full-IBS-TH2.ttl) corresponding tothis example is available.
```
```suggestion
An RDF file containing a [example ttl graph file](Beer-Full-IBS-TH2.ttl) corresponding to this example is available.
```

ldesousa commented on ssn/rdf/examples/readme.md:
```diff
@@ -1,3 +1,55 @@
+
+## Number of sunspots
+
+An RDF file containing a [example ttl graph file](sunspots.ttl) corresponding tothis example is available.
```
```suggestion
An RDF file containing a [example ttl graph file](sunspots.ttl) corresponding to this example is available.
```

ldesousa commented on ssn/rdf/examples/readme.md:
```diff
@@ -1,3 +1,55 @@
+
+## IP68 Smart Sensor
+
+This example describes the IP68 Smart Sensor that and some of its capabilities and operating ranges. A specific IP68 Smart Sensor observes the air temperature, and its own battery state.</p>
```
```suggestion
This example describes the IP68 Smart Sensor and some of its capabilities and operating ranges. A specific IP68 Smart Sensor observes the air temperature, and its own battery state.</p>
```

ldesousa reviewed (dismissed):
I only request changes for the various typos in the Markdown documents. Otherwise, I am fine with merging, even if none of my comments on the RDF are addressed.

dr-shorthair committed:
Merge branch 'gh-pages' into LongExampleAcme

oldskeptic committed:
Update ssn/rdf/examples/readme.md

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

oldskeptic committed:
Update ssn/rdf/examples/readme.md

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

oldskeptic committed:
Update ssn/rdf/examples/readme.md

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

dr-shorthair committed:
Shuffled the intro a bit;
added link to IBS-TH2 description in Complex Systems pattern;
added sosa:startTime and sosa:endTIme to deployments

dr-shorthair committed:
Update ssn/rdf/examples/ACME-Beer-Example.md

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

dr-shorthair committed:
Update ssn/rdf/examples/ACME-Beer-Example.md

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

dr-shorthair committed:
Update ssn/rdf/examples/readme.md

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

dr-shorthair committed:
Update ssn/rdf/examples/ACME-Beer-Example.md

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

dr-shorthair committed:
Update ssn/rdf/examples/ACME-Beer-Example.md

Co-authored-by: Luís de Sousa <luis.de.sousa@protonmail.ch>

dr-shorthair committed:
Various small suggestions from Luis and Ted

dr-shorthair committed:
Merge remote-tracking branch 'origin/LongExampleAcme' into LongExampleAcme

dr-shorthair committed:
Applying further suggestions from TallTed and Luis

dr-shorthair reviewed (approved):
A few minor issues from @ldesousa should still be considered.

dr-shorthair commented:
@oldskeptic a few small outstanding questions from Luis. 

@sgrellet @KathiSchleidt @rob-metalinkage @ldesousa could you look at this pr please? Needs one review.

TallTed commented on ssn/rdf/examples/readme.md:
```diff
@@ -1,20 +1,20 @@
 
 This example is documented in more detail in [ACME-Beer-Example.md](./ACME-Beer-Example.md) 
-An RDF file containing a [example ttl graph file](Beer-Full-IBS-TH2.ttl) corresponding tothis example is available.
+An RDF file containing a [example ttl graph file](Beer-Full-IBS-TH2.ttl) corresponding to this example is available.
```
```suggestion
An RDF file containing an [example Turtle graph file](Beer-Full-IBS-TH2.ttl) corresponding to this example is available.
```

TallTed commented on ssn/rdf/examples/readme.md:
```diff
@@ -1,20 +1,20 @@
 
 ## Tree height measurement
 
 An RDF file containing a [example ttl graph file](tree-height.ttl) corresponding to this example is available.
```
```suggestion
An RDF file containing an [example Turtle graph file](tree-height.ttl) corresponding to this example is available.
```

TallTed commented on ssn/rdf/examples/readme.md:
```diff
@@ -1,20 +1,20 @@
 ## Number of sunspots
 
-An RDF file containing a [example ttl graph file](sunspots.ttl) corresponding tothis example is available.
+An RDF file containing a [example ttl graph file](sunspots.ttl) corresponding to this example is available.
```
```suggestion
An RDF file containing an [example Turtle graph file](sunspots.ttl) corresponding to this example is available.
```

TallTed commented on ssn/rdf/examples/readme.md:
```diff
@@ -48,7 +48,7 @@ An RDF file containing a [example ttl graph file](dht22-deployment.ttl)  corresp
-This example describes the IP68 Smart Sensor that and some of its capabilities and operating ranges. A specific IP68 Smart Sensor observes the air temperature, and its own battery state.</p>
+This example describes the IP68 Smart Sensor and some of its capabilities and operating ranges. A specific IP68 Smart Sensor observes the air temperature, and its own battery state.</p>
 
 An RDF file containing a [example ttl graph file](ip68.ttl)  corresponding to this example is available..
```
```suggestion
An RDF file containing an [example Turtle graph file](ip68.ttl) corresponding to this example is available..
```

ldesousa commented:
> @oldskeptic a few small outstanding questions from Luis.
> 
> @sgrellet @KathiSchleidt @rob-metalinkage @ldesousa could you look at this pr please? Needs one review.

My comments from last week are now outdated. Meanwhile Ted reviewed the text. I am happy if you merge.

dr-shorthair commented:
> > @oldskeptic a few small outstanding questions from Luis.
> > @sgrellet @KathiSchleidt @rob-metalinkage @ldesousa could you look at this pr please? Needs one review.
> 
> My comments from last week are now outdated. Meanwhile Ted reviewed the text. I am happy if you merge.

Thanks @ldesousa - can you review/approve please?

dr-shorthair committed:
Minor changes from @TallTed
